### PR TITLE
Benchmark transaction construction time.

### PR DESCRIPTION
**Pull request by shimkiv**
- Part of the #1506 

---

[Internal issue](https://app.zenhub.com/workspaces/sdk-6130fedb3b0fc600123d8796/issues/zh/92).

### CODE REVIEW

**Review by Trivo25 — APPROVED** [review]

File: benchmark/benchmarks/transaction.ts [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+      }
+    );
+    toc();
+
+    tic('simple zkapp deploy transaction signing');
+    transaction.sign([feePayer.key, zkappPrivateKey]);
+    toc();
+
+    tic('simple zkapp deploy transaction local sending');
+    pendingTransaction = transaction.send();
```
Comment (mitschabaude) [inline benchmark/benchmarks/transaction.ts:69]: this should `await`!

File: benchmark/benchmarks/transaction.ts [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    toc();
+
+    tic('simple zkapp deploy transaction local waiting');
+    await pendingTransaction.wait();
+    toc();
```
Comment (mitschabaude) [inline benchmark/benchmarks/transaction.ts:74]: this is a no-op, so wouldn't include

File: benchmark/benchmarks/transaction.ts [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    await transaction.prove();
+    toc();
+
+    tic('simple zkapp call transaction signing');
+    transaction.sign([feePayer.key]);
+    toc();
+
+    tic('simple zkapp call transaction local sending');
+    pendingTransaction = transaction.send();
+    toc();
```
Comment (mitschabaude) [inline benchmark/benchmarks/transaction.ts:93]: `await`

File: benchmark/benchmarks/transaction.ts [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+    tic('simple zkapp call transaction local waiting');
+    await pendingTransaction.wait();
+    toc();
```
Comment (mitschabaude) [inline benchmark/benchmarks/transaction.ts:97]: remove

**Comment by mitschabaude** [comment]
I'm wondering if proving the transaction is a good idea here 🤔

**Comment by shimkiv** [comment]
> I'm wondering if proving the transaction is a good idea here 🤔

It should not impact much the overall time consumed and we can have the data stored but skip its rendering.

File: benchmark/benchmarks/transaction.ts (hunk 1 above)
Comment (shimkiv) [reply benchmark/benchmarks/transaction.ts:69]: 🤦

**Review by mitschabaude — APPROVED** [review]

**Pull request merged**